### DISCUSSION TITLE
Empty rows with different height than grid rowHeight

### DISCUSSION THREAD

**Discussion by imkitukov** [nick-keller/react-datasheet-grid#285] [Q&A] [answered]
Hey, I need to have the option to use special filler rows(at least visually) to have gaps between grid rows (check the attached picture).

![image](https://user-images.githubusercontent.com/44986252/228946624-b7c361ee-64dd-46ba-b3e9-a20f19d2d839.png)

I tried an approach using value and columns DSG props, where basically those 'filler' rows have special row value (rowData contains rowType: filler and height) and then each column component has the ability to be regulat or 'filler' cell based on rowData.rowType.
The problem is that even I try to set 'filler' cell style to have a certain height (the one from rowData) it turns out DSG rows cant have rows with different heights. Is that correct or I am missing something? Is this doable only with css?

I hope you can suggest another approach as this library looks awesome, but this could be a deal breaker for our project.

**Comment by nick-keller** [comment, marked as answer]
Hey,
It is not documented but you can use a function as your [rowHeight](https://react-datasheet-grid.netlify.app/docs/api-reference/props#rowheight), the function should return the height of the row `((opt: { rowData: T; rowIndex: number }) => number`

```tsx
<DataSheetGrid
    rowHeight={({ rowData }) => rowData.isFiller ? 5 : 40}
/>
```
